components/menu/src/menu-item/menu-item.styles.js

```diff
@@ -88,23 +82,26 @@ export default css`
 
     .label {
         flex-grow: 1;
-        padding: 15px 0;
+        padding: ${spacers.dp12} 0;
     }
 
     li.dense .label {
-        padding: 8px 0;
+        padding: ${spacers.dp8} 0;
     }
 
     .icon {
         flex-grow: 0;
-        margin-right: ${spacers.dp16};
+        margin-right: ${spacers.dp12};
         width: 24px;
         height: 24px;
     }
 
     .chevron {
+        display: flex;
+        align-items: center;
         flex-grow: 0;
-        margin-left: ${spacers.dp48};
+        margin-left: ${spacers.dp24};
+        margin-right: -${spacers.dp12};
```

Comment:
Not sure if this is wise? I couldn't see a way of pulling the `Chevron` icon further to the right without restructuring the component.

Reply:
Negative margins can lead to visual bugs. What's your intention here visually? Do you want it to be further right than this?

![image](https://user-images.githubusercontent.com/7355199/203288105-0fc0fb69-f2c2-4389-ae48-a320e2ec0503.png)

This is what it looks like for me currently, without the negative right margin.

Reply:
It seems there's a default padding of 8px on the `<a>`: https://github.com/dhis2/ui/blob/5cd52efd58424a6bbeb5efe8838fd65814814f9b/components/menu/src/menu-item/menu-item.styles.js#L70-L72 If you want to go right to the right edge of the background, removing that padding will get you there.

The chevron svg has a width of its own, so visually there's still some space left:

![image](https://user-images.githubusercontent.com/7355199/203290280-aec6864b-a87c-4004-b9d7-aa42086d36d7.png)

If you want to move it any further you'll have to tweak the svg styles (setting the viewbox).

Reply:
The existing styles are quite messy in terms of the code btw. We really should clean them up. This goes for a lot of the css of the components in ui. Might be worth setting aside some time to do a thorough refactor of those.

Reply:
Ah, interesting. My intention was to move the chevron to the right when used inside a `FlyoutMenu`. This is how the chevron icon is displayed inside `FlyoutMenu`:
![image](https://user-images.githubusercontent.com/33054985/203307281-f5d28b7c-730c-4607-addc-9e9a62d92b16.png)

So, it looks like the issue is perhaps something to do with the wrong styles being applied. 👀

Reply:
> The existing styles are quite messy in terms of the code btw. We really should clean them up. This goes for a lot of the css of the components in ui. Might be worth setting aside some time to do a thorough refactor of those.

Sounds good. I'm working my way through some minor visual tweaks, now that components have been in use for a while and I've gotten some in-use feedback. Could be good to clean up the styles as part of this process.

Reply:
The `with-chevron` class wasn't properly being applied when a menu-item had children. Fixed this [here](https://github.com/dhis2/ui/pull/1188/files#diff-95560ff80b247b89e0ce8a8eb8fb62109251d8f8ff5a3a9588f2f68780782893R53), and removed the hacky negative margin!

Reply:
> Sounds good. I'm working my way through some minor visual tweaks, now that components have been in use for a while and I've gotten some in-use feedback. Could be good to clean up the styles as part of this process.

Yeah I agree. Maybe we could raise this in our platform meeting and set aside some time for it.

> The with-chevron class wasn't properly being applied when a menu-item had children. Fixed this [here](https://github.com/dhis2/ui/pull/1188/files#diff-95560ff80b247b89e0ce8a8eb8fb62109251d8f8ff5a3a9588f2f68780782893R53), and removed the hacky negative margin!

Ah nice, good detective work! Thanks for the fix!